### PR TITLE
remove deprecated soft_fail from providers part2

### PR DESCRIPTION
**Pull request by raphaelauv**
following : https://github.com/apache/airflow/pull/41710

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: airflow/sensors/time_delta.py [hunk 1]
```diff
@@ -80,12 +79,8 @@ def execute(self, context: Context) -> bool | NoReturn:
...
             # If the target datetime is in the past, return immediately
             return True
-        try:
-            trigger = DateTimeTrigger(moment=target_dttm, end_from_trigger=self.end_from_trigger)
-        except (TypeError, ValueError) as e:
-            if self.soft_fail:
-                raise AirflowSkipException("Skipping due to soft_fail is set to True.") from e
-            raise
+
+        trigger = DateTimeTrigger(moment=target_dttm, end_from_trigger=self.end_from_trigger)
```
Comment (eladkal) [inline airflow/sensors/time_delta.py:83]: wouldn't this cause breaking change in Airflow 2?
https://github.com/apache/airflow/pull/41564 is meant to extract main operators into standard provider. This provider must be supported on Airflow 2 without breaking changes

Comment (eladkal) [reply airflow/sensors/time_delta.py:83]: I think we need to merge the provider PR before this one.

PRs that handle both core and providers together is really hard to navigate. We should avoid this. It just creates confusion

Comment (raphaelauv) [reply airflow/sensors/time_delta.py:83]: my bad I did not saw that `TimeDeltaSensorAsync` was not using `poke` but directly `execute` ( do you know why ? ) 

so yes this operator is not using the soft_fail logic of the BaseSensorOperator

I just did a revert PR -> https://github.com/apache/airflow/pull/41741

Comment (eladkal) [reply airflow/sensors/time_delta.py:83]: Thank you
We can redo the removal on main branch after we release the provider

**Pull request merged**